src/LinearCurve.sol

```diff
@@ -30,80 +30,64 @@ import {BaseCurve} from "./BaseCurve.sol";
+    /// @dev Maximum number of shares that can be handled by the curve.
+    uint256 public constant MAX_SHARES = type(uint256).max;
 
-    /// @dev UD60x18 Max
-    uint256 public immutable MAX_ASSETS = type(uint256).max / 1e18;
+    /// @dev Maximum number of assets that can be handled by the curve.
+    uint256 public constant MAX_ASSETS = type(uint256).max;
 
     /// @notice Constructor for the Linear Curve.
     /// @param _name The name of the curve.
     constructor(string memory _name) BaseCurve(_name) {}
 
     /// @inheritdoc BaseCurve
     /// @notice Computes the 1:1 relationship between assets <--> shares.
-    function previewDeposit(uint256 assets, uint256 totalAssets, uint256 totalShares)
-        public
-        view
+    function previewDeposit(uint256 assets, uint256, /* totalAssets */ uint256 /* totalShares */ )
+        external
+        pure
         override
-        returns (uint256 shares)
+        returns (uint256 /* shares */ )
     {
-        if (assets == 0) return 0;
-
-        shares = assets; // 1:1 relationship
-
-        return shares;
+        return assets; // 1:1 relationship
     }
 
     /// @inheritdoc BaseCurve
     /// @notice Computes the 1:1 relationship between assets <--> shares.
-    function previewRedeem(uint256 shares, uint256 totalShares, uint256 totalAssets)
-        public
-        view
+    function previewMint(uint256 shares, uint256, /* totalShares */ uint256 /* totalAssets */ )
+        external
+        pure
         override
-        returns (uint256 assets)
+        returns (uint256 /* assets */ )
     {
-        if (shares == 0) return 0;
-
-        assets = shares; // 1:1 relationship
-
-        return assets;
+        return shares; // 1:1 relationship
     }
 
     /// @inheritdoc BaseCurve
     /// @notice Computes the 1:1 relationship between assets <--> shares.
-    function previewMint(uint256 shares, uint256 totalShares, uint256 totalAssets)
-        public
-        view
+    function previewWithdraw(uint256 assets, uint256, /* totalAssets */ uint256 /* totalShares */ )
+        external
+        pure
         override
-        returns (uint256 assets)
+        returns (uint256 /* shares */ )
     {
-        if (shares == 0) return 0;
-
-        assets = shares; // 1:1 relationship
-
-        return assets;
+        return assets; // 1:1 relationship
     }
 
     /// @inheritdoc BaseCurve
     /// @notice Computes the 1:1 relationship between assets <--> shares.
-    function previewWithdraw(uint256 assets, uint256 totalAssets, uint256 totalShares)
-        public
-        view
+    function previewRedeem(uint256 shares, uint256, /* totalShares */ uint256 /* totalAssets */ )
+        external
+        pure
         override
-        returns (uint256 shares)
+        returns (uint256 /* assets */ )
     {
-        if (assets == 0) return 0;
-
-        shares = assets; // 1:1 relationship
-
-        return shares;
+        return shares; // 1:1 relationship
```

Comment:
3.21 - Preview Functions in LinearCurve Could Be Simplified

Simplified what was here.